runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -173,6 +180,9 @@
 	@Getter(AccessLevel.PACKAGE)
 	private int points;
 
+	@Getter(AccessLevel.PACKAGE)
+	private Task weaknessTask = null;
```

Comment:
Probably the slayer plugin should always hold a `Task` for the current task instead of specific to the weakness one.

Reply:
This variable is what task the npc the players is interacting with belongs to and if it doesnt belong to one of the tasks that show a weakness overlay it returns null.